zad2/warriors.rb

```diff
@@ -1,12 +1,122 @@
-class Warrior
-  def initialize(name:, level:)
+class Character
+  attr_accessor :name, :level
+
+  #Constructor
+  def initialize(name:,level:)
+    @name = name
+    @level = is_valid?(level)
+  end
+
+  #Card displays all the attributes of our character
+  def card
+  puts "#{name} is #{level} level"
   end
 
+  #Strength method calculates the damage dealt value
   def strength
+    [*1..12].sample + @level
+  end
+
+  protected
+
+  #Finding out if the level is valid (it is supposed to be between 1 and 99)
+  def is_valid?(valid_level)
```

Comment:
+ za walidację, jednak dwie uwagi co do metody `is_valid`:
1. `is_valid?(valid_level)` wygląda na metodę która zwróci `true` / `false` w zależności od tego czy level jest poprawny. nie do końca tak jest, metoda nigdy nie zwróci `false`. Rozwiązaniem mogłoby być nadpisanie levelu w initializerze przez `@level = is_valid?(level) ? level : 1`

2. drobna uwaga dot. nazewnictwa. nazwa argumentu to `valid_level`, ale w momencie startu wykonywania tej metody wcale nie wiadomo czy level jest poprawny :)